include/glow/Backend/Backend.h

```diff
@@ -52,23 +52,29 @@ class Backend {
   /// Generate code for a vector of functions, \p functions. All compilations
   /// use the same settings provided by \p opts. This allows the compiler to
   /// support shared constants between functions.
-  virtual std::vector<std::unique_ptr<CompiledFunction>>
+  virtual llvm::Expected<std::vector<std::unique_ptr<CompiledFunction>>>
```

Comment:
Not sure if it's worth the added error handling for the caller function, but you can allow partial success here by changing the return type to:
_std::vector<llvm::Expected_<std::unique_ptr<CompiledFunction>>>

Reply:
Yeah but imo if any function fails to compile then the whole thing should fail

Reply:
Fair enough :)